File: app/src/main/java/com/mapmyindia/sdk/demo/java/adapter/StyleAdapter.java
package com.mapmyindia.sdk.demo.java.adapter;


import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.databinding.DataBindingUtil;
import androidx.recyclerview.widget.RecyclerView;

import com.bumptech.glide.Glide;
import com.mapmyindia.sdk.demo.R;
import com.mapmyindia.sdk.demo.databinding.LayoutStyleAdapterBinding;
import com.mapmyindia.sdk.maps.style.model.MapmyIndiaStyle;

import java.util.List;

/**
 * Created by Saksham on 05-03-2021
 */

public class StyleAdapter extends RecyclerView.Adapter<StyleAdapter.ViewHolder> {

    private List<MapmyIndiaStyle> styleList;

    private OnStyleSelectListener onStyleSelectListener;

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        LayoutStyleAdapterBinding binding = DataBindingUtil.inflate(LayoutInflater.from(parent.getContext()), R.layout.layout_style_adapter, parent, false);
        return new ViewHolder(binding);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        holder.binding.tvName.setText(styleList.get(position).getDisplayName());
        holder.binding.tvDescription.setText(styleList.get(position).getDescription());
        Glide.with(holder.binding.getRoot().getContext()).load(styleList.get(position).getImageUrl()).into(holder.binding.ivImage);
        holder.binding.getRoot().setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if(onStyleSelectListener != null) {
                    onStyleSelectListener.onStyleSelected(styleList.get(position));
                }
            }
        });
    }

    @Override
    public int getItemCount() {
        return styleList == null? 0: styleList.size();
    }

    public void setStyleList(List<MapmyIndiaStyle> styleList) {
        this.styleList = styleList;
        notifyDataSetChanged();
    }

    public void setOnStyleSelectListener(OnStyleSelectListener onStyleSelectListener) {
        this.onStyleSelectListener = onStyleSelectListener;
    }

    static class ViewHolder extends RecyclerView.ViewHolder {

        private final LayoutStyleAdapterBinding binding;

        public ViewHolder(@NonNull LayoutStyleAdapterBinding itemView) {
            super(itemView.getRoot());
            binding = itemView;
        }
    }

    public interface OnStyleSelectListener {
        void onStyleSelected(MapmyIndiaStyle style);
    }
}

File: app/src/main/java/com/mapmyindia/sdk/demo/kotlin/adapter/StyleAdapter.kt
package com.mapmyindia.sdk.demo.kotlin.adapter

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.databinding.DataBindingUtil
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import com.mapmyindia.sdk.demo.R
import com.mapmyindia.sdk.demo.databinding.LayoutStyleAdapterBinding
import com.mapmyindia.sdk.maps.style.model.MapmyIndiaStyle

/**
 ** Created by Saksham on 24-05-2021.
 **/
class StyleAdapter: RecyclerView.Adapter<StyleAdapter.ViewHolder>() {

    private var styleList: List<MapmyIndiaStyle>? = null
    private var onStyleSelectListener: OnStyleSelectListener? = null


    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val binding: LayoutStyleAdapterBinding = DataBindingUtil.inflate(LayoutInflater.from(parent.context), R.layout.layout_style_adapter, parent, false)
        return ViewHolder(binding)
    }

    override fun getItemCount(): Int {
        return styleList?.size ?: 0
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.binding.tvName.text = styleList?.get(position)?.displayName
        holder.binding.tvDescription.text = styleList?.get(position)?.description
        Glide.with(holder.binding.root.context).load(styleList?.get(position)?.imageUrl).into(holder.binding.ivImage)
        holder.binding.root.setOnClickListener {
            onStyleSelectListener?.onStyleSelected(styleList?.get(position)!!)
        }

    }

    fun setStyleList(styles: List<MapmyIndiaStyle>) {
        styleList = styles
        notifyDataSetChanged()
    }

    fun setOnStyleSelectListener(onStyleSelectListener: OnStyleSelectListener) {
        this.onStyleSelectListener = onStyleSelectListener
    }

    inner class ViewHolder(val binding: LayoutStyleAdapterBinding) : RecyclerView.ViewHolder(binding.root)

    interface OnStyleSelectListener {
        fun onStyleSelected(style: MapmyIndiaStyle)
    }
}